security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -593,6 +566,7 @@ func (sc *SecretManagerClient) rotateTime(secret security.SecretItem) time.Durat
 
 func (sc *SecretManagerClient) registerSecret(item security.SecretItem) {
 	delay := sc.rotateTime(item)
+	item.ResourceName = security.WorkloadKeyCertResourceName
```

Comment:
Both ROOTCA requests and WORKLOAD requests can now force us to request a certificate chain from the CA which gets registered as the secret. While serving the secret, we pick the resource-name from the request.  The "resource-name" stored in the cache is largely ornamental, since the same cache item is serving multiple resource requests. I thought adding this line would prevent confusion.